packages/blade/src/components/TextArea/_decisions/decisions.md

```diff
@@ -0,0 +1,66 @@
+| maxCharacters        | `number`                         |         | Used to show a character counter under the input field                                                                                                                                                                                                       |          |
+| validationState      | `error , success`                |         | Applies the style to the input field based on validation status and shows either errorText or successText respectively                                                                                                                                       |          |
+| helpText             | `string`                         |         | Shown when we want to add some hint to the input field. Displayed under the input field. Only one of helpText, errorText or successText is shown at a time in the priority order as errorText, successText, helpText                                         |          |
+| errorText            | `string`                         |         | Shown when the validationState of the input field is set to Error.                                                                                                                                                                                           |          |
+| successText          | `string`                         |         | Shown when the validationState of the input field is set to Success                                                                                                                                                                                          |          |
+| autoFocus            | `boolean`                        | `false` | The autofocus global attribute is a Boolean attribute indicating that an element should be focused on page load. Web Reference, Native Reference                                                                                                             |          |
+
+
+## Open Questions: 
+
+- [ ] Should the `onClearButton` by default clear the TextArea contents? Or should users handle that logic?
+- [ ] Does the `maxCharacters` just a suggestion or a hard limit? Like if it's set to 20 can users go above 20, like Twitter? 
+- [ ] If the above is true, Should the textarea become invalid when `maxCharacters` count is exceeded? 
+- [ ] How will the `maxLines` behave? In design we set it to minimum of 2 and maximum of 5 lines, when we add `maxLines={4}` will it stay in `2` lines by default and then grow as needed based on the text content? or should it be just `4` lines. If it just stays in `4` lines no matter what then `numberOfLines` or `rows` (native html attribute like) would be a better name.
+- [ ] In design we have that small resize handler on the bottom right corner of the textarea what would happen when user tries to resize the textarea given that we also want to enforce `maxLines`
```

Comment:
Do we also need resize props? Allows resize horizontal / vertical / both?

Reply:
@saurav12 @kamleshchandnani ^

Reply:
> Do we also need resize props? Allows resize horizontal / vertical / both?

@divyanshu013 @anuraghazra Yes, we need it so that users can customize the height & width. This helps with the overall accessibility.

Reply:
we discussed to descope this for now because of trailing visual elements and it's visual issues